### PR TITLE
rustcat 3.0.0 (new formula)

### PR DESCRIPTION
**Pull request by 7a6163**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

File: Formula/r/rustcat.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  desc "Modern Port listener and Reverse shell"
+  homepage "https://github.com/robiot/rustcat"
+  url "https://github.com/robiot/rustcat/archive/refs/tags/v3.0.0.tar.gz"
+  sha256 "59648e51ab41e4aeb825174dfbb53710207257feb3757521be98ed28c4249922"
+  license "GPL-3.0-or-later"
+
+  livecheck do
+    url :stable
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
```
Comment (ZhongRuoyu) [inline Formula/r/rustcat.rb:11]: We don't usually need an explicit `livecheck` block unless livecheck gives incorrect results.

```suggestion
```

File: Formula/r/rustcat.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class Rustcat < Formula
+  desc "Modern Port listener and Reverse shell"
+  homepage "https://github.com/robiot/rustcat"
+  url "https://github.com/robiot/rustcat/archive/refs/tags/v3.0.0.tar.gz"
+  sha256 "59648e51ab41e4aeb825174dfbb53710207257feb3757521be98ed28c4249922"
+  license "GPL-3.0-or-later"
```
Comment (ZhongRuoyu) [inline Formula/r/rustcat.rb:6]: Where does the "or-later" part come from? It seems only `GPL-3.0` is specified: https://github.com/robiot/rustcat/blob/c9e20dde398ee3a8655f95c8f6a8e3fbbb6aee60/Cargo.toml#L6

File: Formula/r/rustcat.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    port = free_port
+
+    r, _, _pid = PTY.spawn("#{bin}/rcat listen #{port}")
+    output = r.readline.gsub(/\e\[[0-9;]*m/, "")
+    assert_match "info: Listening on 0.0.0.0:#{port}", output
```
Comment (ZhongRuoyu) [inline Formula/r/rustcat.rb:24]: We need to clean up the spawned process.

```suggestion
    r, _, pid = PTY.spawn("#{bin}/rcat listen #{port}")
    output = r.readline.gsub(/\e\[[0-9;]*m/, "")
    assert_match "info: Listening on 0.0.0.0:#{port}", output
  ensure
    Process.kill("TERM", pid)
    Process.wait(pid)
  end
```

File: Formula/r/rustcat.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+class Rustcat < Formula
+  desc "Modern Port listener and Reverse shell"
+  homepage "https://github.com/robiot/rustcat"
+  url "https://github.com/robiot/rustcat/archive/refs/tags/v3.0.0.tar.gz"
+  sha256 "59648e51ab41e4aeb825174dfbb53710207257feb3757521be98ed28c4249922"
+  license "GPL-3.0-only"
```
Comment (chenrui333) [inline Formula/r/rustcat.rb:6]:
```suggestion
  license "GPL-3.0-only"
  head "https://github.com/robiot/rustcat.git", branch: "master"
```

**Review by chenrui333 — COMMENTED** [review]
can you also add rustcat into autobump list as well? thanks!

File: Formula/r/rustcat.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
```
Comment (ZhongRuoyu) [inline Formula/r/rustcat.rb:6]: Sorry, maybe I wasn't clear: `GPL-3.0` does not imply `GPL-3.0-only`. Can you ask upstram to confirm whether later GPL licenses apply to this software? Thanks!

Comment (7a6163) [reply Formula/r/rustcat.rb:6]: https://github.com/robiot/rustcat/issues/65

@chenrui333 has already inquired with the author, awaiting a reply.

Comment (robiot) [reply Formula/r/rustcat.rb:6]: @7a6163 you can use GPL-3.0-only here. I will update the cargo.toml, but I can't publish a new version atm.

Comment (ZhongRuoyu) [reply Formula/r/rustcat.rb:6]: Thanks!

**Review by ZhongRuoyu — APPROVED** [review]
Thank you @7a6163 for your first contribution to `homebrew/core`!

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**